node/src/types/deploy.rs

```diff
@@ -833,6 +833,11 @@ impl Deploy {
             None,
         )
     }
+
+    #[cfg(test)]
+    pub(crate) fn invalidate(&mut self) {
```

Comment:
IMO this should be a function and not a method. How do you know that there's no chain whose `chain_name == ""`? It's a rather unique way to invalidate a deploy and is probably use-case specific.